doc/poe/poe_hld.md

```diff
@@ -0,0 +1,370 @@
+- config poe interface power_limit \<ifname\> {power_limit}
+
+Examples:
+```
+$ config poe interface status Ethernet0 enable
+$ config poe interface priority Ethernet0 high
+$ config poe interface power_limit Ethernet0 20.4
+```
+
+**TODO**: update CLI reference https://github.com/sonic-net/sonic-utilities/blob/master/doc/Command-Reference.md
+
+#### YANG model Enhancements
+
+TBD
+
+#### Config DB Enhancements  
+
+New table has been added to the config database to store related PoE configuration parameters:
+
+##### POE_PORT_TABLE
+~~~
+    ;Stores PoE port configuration
+    key         = POE_PORT|ifname         ; ifname with prefix POE_PORT
+    ; field     = value
+    enabled     = "enable" / "disable"    ; enable/disable PoE on port, default "disable"
+    power_limit = 1*3.1DIGIT              ; power limit on PoE port
+    priority    = "crit" / "high" / "low" ; port priority level, default "high"
+~~~
+
+#### Application DB Enhancements  
+
+In SONiC all the peripheral devices data (information relared to PoE) will be stored in application database in separated tables:
+
+#### POE_PSE_STATE_TABLE
+~~~
+    ; Defines information for a global PoE state
+    key         = POE_TABLE:GLOBAL   ; key
+    ; field     = value
+    platform_supported = BOOLEAN                             ; PoE platform support
+    poe_status         = "N/A" / "init" / "failed" / "ready" ; PoE status
+    total_ports        = 1*2DIGIT                            ; total PoE ports
+    total_pwr          = 1*4.3DIGIT                          ; total power
+    pwr_consump        = 1*4.3DIGIT                          ; total power consumption
+    pwr_avail          = 1*4.3DIGIT                          ; available power
+    pse_sw_ver         = STRING                              ; PSE software version
+    pse_hw_ver         = STRING                              ; PSE hardware version
+    pse_temp           = 1*3DIGIT                            ; PSE temperature
+    pse_status         = "active" / "fail" / "not present"   ; PSE status
+~~~
+
+#### POE_PORT_STATE_TABLE
+    ; Defines information for a PoE port state
+    key         = POE_PORT_STATE|ifname   ; ifname with prefix POE_PORT_STATE
+    ; field     = value
+    fp_port     = 1*2DIGIT                ; PoE front panel port (map PoE port num to ifname)
+    status      = BOOLEAN                 ; presence state of the PoE
+    enabled     = "enable" / "disable"    ; PoE enabled on port
+    priority    = "crit" / "high" / "low" ; port priority level
+    protocol    = STRING                  ; name of PoE protocol
+    class       = 1*DIGIT                 ; power class
+    pwr_consump = 1*3.3DIGIT              ; the power of the PoE
+    pwr_limit   = 1*3.3DIGIT              ; power capacity of the PoE port
+    voltage     = 1*3.3DIGIT              ; output voltage of the PoE port
+    current     = 1*3.3DIGIT              ; the current of the PoE port
+
+
+### Warmboot and Fastboot Design Impact  
```

Comment:
Please capture if we are planning to support persistent supply of power to PD from PSE during reboot.

Reply:
It's already in the current design.